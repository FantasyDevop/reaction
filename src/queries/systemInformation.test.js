import mockContext from "@reactioncommerce/api-utils/tests/mockContext.js";
import ReactionError from "@reactioncommerce/reaction-error";
import systemInformation from "./systemInformation.js";

const fakeShopId = "FAKE_SHOP_ID";
mockContext.validatePermissions = jest.fn("validatePermissions");
// mockContext.app = {};
mockContext.app = { db: { admin: jest.fn("admin") } };

beforeEach(() => {
  jest.resetAllMocks();
});

test("throws access-denied if not allowed", async () => {
  mockContext.validatePermissions.mockImplementation(() => {
    throw new ReactionError("access-denied", "Access Denied");
  });
  const result = systemInformation(mockContext, fakeShopId);
  return expect(result).rejects.toThrowErrorMatchingSnapshot();
});
