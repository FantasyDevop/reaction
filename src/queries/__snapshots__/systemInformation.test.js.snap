// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws access-denied if not allowed 1`] = `"Cannot destructure property \`db\` of 'undefined' or 'null'."`;
